### ISSUE TITLE
Post List: On retry media upload sometimes fails but the post gets published

### ISSUE THREAD

**Issue by malinajirka** [wordpress-mobile/WordPress-Android#9704]
### Expected behavior
When uploading media fails, clicking on Retry button will upload the media and upload the post.

### Actual behavior
When uploading media fails, clicking on Retry button **sometimes** uploads the post, but the media are not uploaded. Actually it appears the media got uploaded, as they are displayed in the app, but the path to the file isn't correct -> it contains path to the local file eg. "/storage/emulated/...".

### Steps to reproduce the behavior
1. Click on create post button
2. Enter a title
3. Set Network to GPRS on the emulator
4. Add an image from the local storage
5. Click on Publish
6. Restart the app - Kill it so even the services are killed (swiping it away from the recent apps view isn't enough - I use adb plugin)
7. Go to My Site -> Post list
8. Notice "error uploading media" label is shown
9. Set Network back to LTE
10. Click on retry
11. Wait until the upload completes
12. Open the post
13. Switch to html view
14. Notice the path contains a path to the local storage

![retry-media-upload](https://user-images.githubusercontent.com/2261188/56660929-e8ea9900-66a0-11e9-836e-913ba7e0cb5b.gif)

##### Tested on [device], Android [version], WPAndroid [version]
Emulator API 27

**Comment by shiki** [comment]
Possibly related to #9804.

**Comment by malinajirka** [comment]
Possibly related to https://github.com/wordpress-mobile/WordPress-Android/issues/9802

**Comment by maxme** [comment]
I spent way too much time trying to debug this :)

Some notes:
- I wasn't able to reproduce it with 1 image.
- I was able to reproduce it with 2 images before the patch https://github.com/wordpress-mobile/WordPress-Android/pull/9859
- I wasn't able to reproduce it with 2 images after the patch https://github.com/wordpress-mobile/WordPress-Android/pull/9859

Theory: that bug can't happen now with #9859 merged because image ids are correctly associated to the failed upload post, so when we hit retry, media uploads are re-tried and the post upload will fail if the media aren't uploaded. This prevent a post to be published with failed media. Prior to #9859 it was possible because image ids were not correctly associated to the post in certain cases (multi selection in the editor).
